components/external_media_links/external_media_link.lua

```diff
@@ -0,0 +1,270 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:ExternalMediaLink
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Flags = require('Module:Flags')
+local Logic = require('Module:Logic')
+local Page = require('Module:Page')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local ExternalMediaLink = {}
+
+local MAXIMUM_VALUES = {
+	subjects = 20,
+	organisations = 5,
+	authors = 5,
+}
+local DEFAULT_LANGUAGE = 'en'
+local NON_BREAKING_SPACE = '&nbsp;'
+
+function ExternalMediaLink.run(args)
+	ExternalMediaLink._fallBackArgs(args)
+
+	if Logic.nilOr(Logic.readBoolOrNil(args.storage), true)
+		and not Logic.readBool(Variables.varDefault('disable_LPDB_storage'))
+		and not Logic.readBool(Variables.varDefault('disable_SMW_storage')) then
+
+		ExternalMediaLink._store(args)
+	end
+
+	mw.ext.TeamLiquidIntegration.add_category('Pages with ExternalMediaLinks')
+	if not Logic.nilOr(Logic.readBoolOrNil(args.display), true) then
+		return
+	end
+
+	return ExternalMediaLink._display(args)
+end
+
+function ExternalMediaLink._fallBackArgs(args)
+	args.by_link1 = args.by_link1 or args.by_link
+	args.by1 = args.by1 or args.by
+	args.subject1 = args.subject1 or args.subject or args.player or args.interviewee
+	args.subject_organization1 = args.subject_organization
+
+	for subjectIndex = 2, MAXIMUM_VALUES.subjects do
+		args['subject' .. subjectIndex] = args['subject' .. subjectIndex] or args['player' .. subjectIndex]
+	end
+end
+
+function ExternalMediaLink._store(args)
+	local lpdbData = {
+		date = args.date,
+		language = args.language or DEFAULT_LANGUAGE,
+		title = args.title,
+		translatedtitle = args.trans_title,
+		link = args.link,
+		publisher = args.of,
+		type = args.type,
+	}
+
+	local authors = {}
+	for authorIndex, author in Table.iter.pairsByPrefix(args, 'by') do
+		authors['author' .. authorIndex] = args['by_link' .. authorIndex] or author
```

Comment:
Still something i'd like to have would be resolving redirects to be able to accomodate pagemoves.

Reply:
```suggestion
		authors['author' .. authorIndex] = mw.ext.TeamLiquidIntegration.resolve_redirect(args['by_link' .. authorIndex] or author)
```